src/Microsoft.DotNet.ApiCompat/Program.cs

```diff
@@ -17,148 +17,212 @@
 using Microsoft.Cci.Filters;
 using Microsoft.Cci.Mappings;
 using Microsoft.Cci.Writers;
-using Microsoft.Cci.Writers.Syntax;
-using Microsoft.Fx.CommandLine;
 using System.Reflection;
+using McMaster.Extensions.CommandLineUtils;
 
 namespace Microsoft.DotNet.ApiCompat
 {
-    public class ExportCciSettings
+    public class Program
     {
-        public static IEqualityComparer<ITypeReference> StaticSettings { get; set; }
-        public static IDifferenceOperands  StaticOperands { get; set; }
-        public static IAttributeFilter StaticAttributeFilter { get; set; }
-
-        public ExportCciSettings()
+        public static int Main(string[] args)
         {
-            Settings = StaticSettings;
-            Operands = StaticOperands;
-            AttributeFilter = StaticAttributeFilter;
-        }
+            var app = new CommandLineApplication
+            {
+                Name = "ApiCompat",
+                FullName = "A command line tool to verify that two sets of APIs are compatible.",
+                ResponseFileHandling = ResponseFileHandling.ParseArgsAsSpaceSeparated
+            };
+            app.HelpOption("-?|-h|--help");
+            app.VersionOption("-v|--version", GetAssemblyVersion());
+
+            CommandArgument contracts = app.Argument("contracts", "Comma delimited list of assemblies or directories of assemblies for all the contract assemblies.");
+            contracts.IsRequired();
+            CommandOption implDirs = app.Option("--implDirs", "Comma delimited list of directories to find the implementation assemblies for each contract assembly.", CommandOptionType.SingleValue);
+            implDirs.IsRequired(allowEmptyStrings: true);
+            CommandOption listRules = app.Option("--listRules", "Outputs all the rules. If this options is supplied all other options are ignored.", CommandOptionType.NoValue);
+            CommandOption baseline = app.Option("--baseline", "Baseline file to skip known diffs.", CommandOptionType.SingleValue);
+            CommandOption remapFile = app.Option("--remapFile", "File with a list of type and/or namespace remappings to consider apply to names while diffing.", CommandOptionType.SingleValue);
+            CommandOption skipGroupByAssembly = app.Option("--skipGroupByAssembly", "Skip grouping the differences by assembly instead of flattening the namespaces.", CommandOptionType.NoValue);
```

Comment:
As long as we're changing this, should we update this to follow POSIX casing conventions for options?
```suggestion
            CommandOption skipGroupByAssembly = app.Option("--skip-group-by-assembly", "Skip grouping the differences by assembly instead of flattening the namespaces.", CommandOptionType.NoValue);
```